### PR TITLE
job agentPreOfferingDelete and agentPreOfferingPopUp

### PR DESCRIPTION
**Pull request by furkhat**

### CODE REVIEW

File: proc/worker/agent.go [hunk 1]
```diff
@@ -597,3 +597,107 @@ func (w *Worker) AgentAfterOfferingDelete(job *data.Job) error {
...
+	if err != nil {
+		return nil, err
+	}
+
+	if offering.OfferStatus != data.OfferRegister {
+		return nil, fmt.Errorf("offering is not registered")
+	}
+
+	var pendingJobs int
+	row := w.db.QueryRow(`SELECT COUNT(*)
```
Comment (ababo) [inline proc/worker/agent.go]: I'd suggest to format SQL [properly](https://www.sqlstyle.guide/).

File: proc/worker/agent.go [hunk 2]
```diff
@@ -597,3 +597,107 @@ func (w *Worker) AgentAfterOfferingDelete(job *data.Job) error {
...
+	tx, err := w.ethBack.PSCPopupServiceOffering(auth, offeringHash)
+	if err != nil {
+		return err
+	}
+
+	return w.saveEthTX(job, tx, "PopupServiceOffering", job.RelatedType,
+		job.RelatedID, offering.Agent, data.FromBytes(w.pscAddr.Bytes()))
+}
+
+func (w *Worker) validateOfferingBeforeMutation(job *data.Job, jType string) (*data.Offering, error) {
```
Comment (ababo) [inline proc/worker/agent.go]: Can we keep under 80 characters per line?

File: uisrv/offering.go [hunk 3]
```diff
@@ -210,14 +210,26 @@ func (s *Server) handlePutOfferingStatus(
 	if !s.parsePayload(w, r, req) {
 		return
 	}
-	// TODO: popup, deactivate
-	if req.Action != PublishOffering {
+
+	var jobType string
+	if req.Action == PublishOffering {
+		jobType = data.JobAgentPreOfferingMsgBCPublish
+	}
```
Comment (ababo) [inline uisrv/offering.go]: Please add `else`s.

File: proc/worker/agent.go [hunk 4]
```diff
@@ -597,3 +597,107 @@ func (w *Worker) AgentAfterOfferingDelete(job *data.Job) error {
...
+	if err != nil {
+		return nil, err
+	}
+
+	if offering.OfferStatus != data.OfferRegister {
+		return nil, fmt.Errorf("offering is not registered")
+	}
+
+	var pendingJobs int
+	row := w.db.QueryRow(
```
[the remark this answers is not in the document]
Comment (furkhat) [reply proc/worker/agent.go]: It depends whether we want this rule to be strict or not. By setting `Duplicated = true` in the config file, we can allow duplicates. 
I vote for rule to be strict.

**Pull request merged**